### DISCUSSION TITLE
Error message: "Deployment template validation failed is not defined in the template"

### DISCUSSION THREAD

**Discussion by mapetnik** [Azure/avdaccelerator#535] [Q&A]
Dear Dany,

I am still facing an issue during deployment the AVD alerts template. My AVD resources does not reside in just one resource group. No matter, if I am deploying the Alerts template using Azure UI or by PowerShell, it's still not working. I have to had to fixed some issues, but at the end I am still getting this error message. I guess, some dependings is missing and I have checked a lot of them, but anyway:

"Message=Deployment template validation failed: 'The resource 'Microsoft.Resources/deployments/edu-esm-rg-log' is not defined in the 
template. Please see https://aka.ms/arm-syntax for usage details.'."

Meanwhile I have spent a lot of my time but I am not able to get it runninig. could you give me some advice please. In case you need my template/parameter file just let me know.

Thanks in advance and I look forward to hearing from you soon.

Best regards,
Sebastian

**Comment by danycontre** [comment]
@mapetnik thanks for your feedback, we are checking.

@JCoreMS

**Reply by mapetnik** [reply]
Good morning Dany,

Just would like to know, how is the progress of my previous request? And In case you need my template files let me know please.

-> Beyond that during deploying the MSIX App tool vm by ARM I am also facing a further issue:

Status Message: Unable to process template
     | language expressions for resource
     | '/subscriptions/xxxxea2-xxx-4f68-xxxx-2114f4badc12/resourceGroups/xxx-xxx-rg-msix/providers/Microsoft.Resources/deployments/linked_xxx-xxx-vm-msix_deployment' at line '136' and column '9'. 'Unable to evaluate template language function 'extensionResourceId': function requires exactly two multi-segmented arguments. 

-> Apart from that the Deep Insights Workbook guide doesn't work as well. After adding the JSON link, the dashboard is not showing anything. It's just blank. I guess the most of that features have already been added to AVD Insights. I recognized many new thresholds and helpful informations that I was mssing until now.

Thank you for you support I am really appreciate your help.
Best regards,
Sebastian

**Reply by mapetnik** [reply]
Dear Dany,

meanwhile I could fix this issue "extensionResourceId" on my own but I still need support by the "Deployment template validation" issue.
Can you give an estimation, when will you be able to provide some feedback related to my error?

Best regards,
Sebastian

**Comment by mapetnik** [comment]
Hello Dany,
Any feedback related to my previous question: Error message: "Deployment template validation failed is not defined in the template"?
Thank you in advance and best regards,
Sebastian

**Comment by danycontre** [comment]
@mapetnik Sorry we missed this one, we are monitoring issues but not the discussions.

Are you still having the issue?

**Comment by mapetnik** [comment]
Hi, yes I could fix it by a workaround but this took me long of time to find the working way.

**Comment by mapetnik** [comment]
Hi Deny, 

I noticed two issues/hints:
- The created Automation account is not granted at the host pool by deployment just only at the storage account level. Only when I assigned to it the following RBAC rights "Desktop Virtualization Host Pool Contributor" I could see an output in the host pool runbook (please see attachment). 
- Quering the the Analytics workspace I can not find the sent outputs (please see attachment). There are no any jobs fired to my distribution group.

![LogAnalyticsWorkspace](https://github.com/Azure/avdaccelerator/assets/25727047/b6770798-c5b0-469b-9d4a-e521e526980a)
![AVDHostPoolLogData](https://github.com/Azure/avdaccelerator/assets/25727047/6b1c36b6-d19b-4bb8-95e6-17453b083bbd)

Thanks for investigating my problems.
Best regards,
Sebastian

**Reply by danycontre** [reply]
Thanks, we are checking.

@swathibhat1 @moisesjgomez

**Reply by JCoreMS** [reply]
@mapetnik 
Not sure how I missed this but here is the gist of a few of the alerts as noted in the readme. 
AVD-HostPool-Capacity  
AVD-Storage-Low Space on Azure File Share-X% Remaining-{volumename}  

These alerts use the query that you noted above which is populated via the automation account created which has an associated runbook. This was done due to information we don't get from the AVD agents.  However I've also done some updates as of Oct or Nov 2023 that should have fixed this issue.  Have you tried to re-deploy with the updated solution?

**Comment by danycontre** [comment]
@roarrioj

**Comment by mapetnik** [comment]
Hi both, thanks for your reply. I could fix all of my issues on my own. I think, I'm fine for the moment thank you.
